runners/direct-java/src/main/java/org/apache/beam/runners/direct/WriteWithShardingFactory.java

```diff
@@ -61,10 +63,11 @@
       AppliedPTransform<PCollection<InputT>, PDone, PTransform<PCollection<InputT>, PDone>>
           transform) {
     try {
+      List<PCollectionView<?>> sideInputs =
+          WriteFilesTranslation.getDynamicDestinationSideInputs(transform);
+      FileBasedSink sink = WriteFilesTranslation.getSink(transform);
       WriteFiles<InputT, ?, ?> replacement =
-          WriteFiles.to(
-              WriteFilesTranslation.getSink(transform),
-              WriteFilesTranslation.getFormatFunction(transform));
+          WriteFiles.to(sink).withOverriddenSideInputs(sideInputs);
```

Comment:
Done - also I changed WriteFiles so this is now a normal codepath, not something just for surgery